NOTE: I couldn't open the professor's notebook in my machine, so I am including my answers in a separate one.

In [13]:
import numpy as np

In [28]:
#Z-test

alpha=0.05

#Converting to fractions
P_0=52.1*0.01
P_1=53.3*0.01

n_0=409
n_1=564

#Getting the counts
Nt_0=P_0*n_0
Nt_1=P_1*n_1

#Defining sample proportions
sp=(P_0*n_0+P_1*n_1)/(n_1+n_0)

#Calculating standard deviation
sp_stdev= lambda p, n: np.sqrt( p * ( 1 - p ) /n[0] +  p * ( 1 - p )/n[1]  )
sp_stdev_2y=sp_stdev((Nt_0+Nt_1)/(n_0+n_1),[n_0,n_1])
print P_0, n_0, n_1, sp_stdev_2y

0.521 409 564 0.0324224126866


In [30]:
#Calculating Z-score
zscore = lambda p0, p1, s : (p0-p1)/s
z_2y = zscore(P_1, P_0, sp_stdev_2y)
print z_2y

0.370114343926


In [32]:
p_2y=1-0.6443

def report_result(p,a):
    print 'is the p value {0:.2f} smaller than the critical value {1:.2f}? '.format(p,a)
    if p<a:
        print "YES!"
    else: print "NO!"
    
    print 'the Null hypothesis is {}'.format( 'rejected' if p<a  else 'not rejected') 

    
report_result(p_2y,alpha)

is the p value 0.36 smaller than the critical value 0.05? 
NO!
the Null hypothesis is not rejected


In [11]:
#Chi-square

Ntot = 973
expected = 564*409*((14.7*5.64)+(11.9*4.09))*((85.3*5.64)+(88.1)*(4.09))
sample_values = [[14.7*5.64,85.3*5.64],[11.9*4.09,88.1*4.09]]
 
chisqstat= lambda N, values, expect : N*((values[0][0]*values[1][1]-values[0][1]*values[1][0])**2)/(expect)

print chisqstat(Ntot,  sample_values, expected)

1.58939333193


COMMENTS: 3.84 is mentioned in the example because that is the corresponding value in the Chi-square table to 1 degree of freedom and an alpha of 0.05. The statistic I calculated of 1.59 is lower than 3.84, leading to the rejection of the hypothesis that the variables are independent (i.e. it corroborates the efficacy of the program).